sky/utils/schemas.py

```diff
@@ -517,8 +517,15 @@ def get_cluster_schema():
 
 _REMOTE_IDENTITY_SCHEMA = {
     'remote_identity': {
-        'type': 'string',
-        'case_insensitive_enum': ['LOCAL_CREDENTIALS', 'SERVICE_ACCOUNT'],
+        'oneOf': [{
+            'type': 'string'
+        }, {
+            'type': 'object',
+            'required': [],
+            'additionalProperties': {
+                'type': 'string',
+            },
+        }]
```

Comment:
Seems we only added the support for AWS at the moment, we should limit this ability to `aws` cloud only, i.e. in L664, we can only apply this additional support for `aws` only.

Reply:
Should be fixed